### PR TITLE
 NOISSUE - Add: View domain permissions

### PR DESCRIPTION
**Pull request by arvindh123**
# What type of PR is this?

## Which issue(s) does this PR fix/relate to?

**Add New endpoint to view domain permissions**

Example: 

Request : 
```bash
curl --location 'http://localhost/domains/3e569c55-3c94-4923-93c0-06ea00a9927a/permissions' \
--header 'Authorization: Bearer <TOKEN>'
```
Response:
```json
{
    "permissions": [
        "admin",
        "edit",
        "view",
        "membership"
    ]
}
```

- Related Issue #
- Resolves #

### CODE REVIEW

**Comment by ianmuchyri** [comment]
@arvindh123 , I think it would be easier if we listed the permissions instead of the relations. Sth similar to how you implemented this: https://github.com/absmach/magistrala/pull/66

**Review by ianmuchyri — DISMISSED** [review]
LGTM

File: auth/postgres/domains.go [hunk 1]
```diff
@@ -92,6 +92,35 @@ func (repo domainRepo) RetrieveByID(ctx context.Context, id string) (auth.Domain
...
+func (repo domainRepo) RetrievePermissions(ctx context.Context, subject, id string) ([]string, error) {
+	q := fmt.Sprintf(`SELECT pc.relation as relation
+	FROM domains as d
+	JOIN policies pc
+	ON pc.object_id = d.id
+	WHERE d.id = '%s'
+	AND pc.subject_id = '%s'
+	`, id, subject)
+
+	rows, err := repo.db.QueryxContext(ctx, q)
```
Comment (dborovcanin) [inline auth/postgres/domains.go:104]: Let's use a named query (`where d.id = :name`) or prepared statement here (`where d.id = $1...`).

**Comment by ianmuchyri** [comment]
Add SDK function for domainPermissions

**Review by dborovcanin — CHANGES_REQUESTED** [review]

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**